frontend/packages/pipelines-plugin/src/components/taskruns/useTaskRuns.ts

```diff
@@ -0,0 +1,36 @@
+import { k8sListResource } from '@console/dynamic-plugin-sdk/src/utils/k8s';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { TaskRunModel } from '../../models';
+import { TaskRunKind } from '../../types';
+import { TektonResourceLabel } from '../pipelines/const';
+
+export const useTaskRuns = (
+  namespace: string,
+  pipelineRunName: string,
+): [TaskRunKind[], boolean, unknown] =>
+  useK8sWatchResource<TaskRunKind[]>({
+    kind: referenceForModel(TaskRunModel),
+    namespace,
+    selector: {
+      matchLabels: {
+        [TektonResourceLabel.pipelinerun]: pipelineRunName,
```

Comment:
Do we ever use this hook on a page that shows the status for multiple pipeline runs? I worry about cases where we could open lots of watches to calculate the status (e.g., one for every table row).

Reply:
If I'm correct its per row. We have this follow up PR where we can check the performance: [OCPBUGS-13152](https://issues.redhat.com/browse/OCPBUGS-13152)